js/qz-tray.js

```diff
@@ -1988,10 +2011,44 @@ var qz = (function() {
              *
              * @param {Function} promiseGen <code>Function({function} toSign)</code> Should return a function, <code>Function({function} resolve)</code>, that
              *     will sign the content and resolve the created promise.
+             *
              * @memberof qz.security
              */
             setSignaturePromise: function(promiseGen) {
                 _qz.security.signaturePromise = promiseGen;
+            },
+
+            /**
+            * Set which signing algorithm QZ will check signatures against.
+            *
+            * @param {string} algorithm The algorithm used in signing. Valid values: <code>[SHA1 | SHA256 | SHA512]</code>
+            * @since 2.1.0
+            *
+            * @memberof qz.security
+            */
+            setSignatureAlgorithm: function(algorithm) {
+                //warn for incompatibilities if known
+                if (!_qz.security.isAlgorithmCompatible()) {
+                    return;
```

Comment:
@tresf, I'm not sure if preventing setting the algorithm as well is the right choice or if we should just make the call to print a warning and let it be set anyway.
And if we do want to prevent it, then in cases where the algorithm was set before a connection, should we reset it back to sha1 or just let it be?

Reply:
It's informational, so I think a soft warning is fine for now.

Reply:
On second thought, maybe we shouldn't be checking if it's compatible here?  We usually setup the API before connection, so this software warning will almost never fire.  Can we refactor this to a location that will fire 100% of the time?

Reply:
We call `isAlgorithmCompatible` here and [after opening](https://github.com/qzind/tray/pull/525/files#diff-5b383df0ac6d733ed512ffe910a83668R343) a connection

Reply:
Perfect.  Merging.